packages/cdktf-cli/bin/cmds/helper/outputs.ts

```diff
@@ -0,0 +1,47 @@
+import * as fs from "fs";
+import * as path from "path";
+import { TerraformOutput, isTerraformOutput } from "../ui/models/terraform";
+import { NestedTerraformOutput } from "../ui/terraform-context";
+
+export type Outputs = { [key: string]: TerraformOutput };
+export type OutputIdMapLeaf = { [constructId: string]: string };
+export type OutputIdMapNode = { [stackOrConstructId: string]: OutputIdMap };
+export type OutputIdMap = OutputIdMapLeaf | OutputIdMapNode;
+
+function unpackTerraformOutput(
+  outputs: NestedTerraformOutput,
+  includeSensitiveOutputs: boolean
+): Record<string, string> {
+  return Object.entries(outputs).reduce(
+    (acc, [key, entry]) => ({
+      ...acc,
+      [key]: isTerraformOutput(entry)
+        ? !entry.sensitive || includeSensitiveOutputs
+          ? entry.value
+          : "<sensitive>"
+        : unpackTerraformOutput(entry, includeSensitiveOutputs),
+    }),
+    {}
+  );
+}
+
+export async function saveOutputs(
+  filePath: string,
+  outputs: NestedTerraformOutput,
+  includeSensitiveOutputs: boolean
+) {
+  fs.writeFileSync(
+    filePath,
+    JSON.stringify(
+      unpackTerraformOutput(outputs, includeSensitiveOutputs),
```

Comment:
This will print `<sensitive>` if `includeSensitiveOutputs` is `false`, correct?
I'd rather have expected that sensitive outputs don't appear at all in the output file in this case. If an output is switched to a sensitive value (i.e. for a previously hard-coded password) it could cause trouble if the user does not notice the change in the contents of the file and deploys services based on the password "<sensitive>" 😅

Reply:
Yeah that makes sense to me 👍